dialer.go

```diff
@@ -27,15 +34,36 @@ type Dialer struct {
 	configPB   []byte
 }
 
-func NewDialer() *Dialer {
+func NewDialer(is_ios bool) *Dialer {
 	water.SetGlobalCompilationCache(compilationCache)
 
-	return &Dialer{
+	var dialer = Dialer{
 		protectedDial: func(network, address string) (net.Conn, error) {
 			return nil, ErrNoDialer
 		},
 		unprotectedDial: net.Dial,
 	}
+
+	if is_ios {
+		// Create a ConfigJSON object with only ForceInterpreter set
+		config := ConfigJSONiOS{
+			Runtime: struct {
+				ForceInterpreter bool `json:"force_interpreter,omitempty"`
+			}{
+				ForceInterpreter: true, // Setting this to true or false as needed
+			},
+		}
+
+		// Marshal the config object to JSON
+		jsonData, err := json.Marshal(config)
+		if err != nil {
+			panic(fmt.Sprintf("Error marshaling JSON:", err))
+		}
+
+		dialer.configJSON = jsonData
+	}
```

Comment:
This change looks redundant. Should it be possible for us to programmatically detect iOS and switch `ForceInterpreter` to `true` without using the JSON (JSON was meant to be reserved for out-sourced customization).

Reply:
For now I inserted a bool to distinguish iOS and Android in this API (`NewDialer`) from the Go part. And we can definitely eliminating the usage of JSON to set this where I can move this bool to `dialWATER` where we can first check if we have customized JSON / Proto to set things, otherwise if it's from iOS, then we create with the interpreter mode there.

Reply:
I was thinking about something else. Like, we can base on build tag to set a flag to true for iOS and directly call `*(water.Config).RuntimeConfig().Interpreter()`

Reply:
I will try to push a change reflecting what I described.

Reply:
Thanks for the changes, I think it looks great!